### PR TITLE
Setting the display of constant layers in the exec graph

### PR DESCRIPTION
**Pull request by egorshul**
### Details:
 - *Added the ability not to display constant layers in exec graph*

### Tickets:
 - *29797*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky please review the PR

File: inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp [hunk 1]
```diff
@@ -119,22 +119,34 @@ InferenceEngine::CNNNetwork dump_graph_as_ie_ngraph_net(const MKLDNNGraph &graph
...
     ngraph::NodeVector to_hold;
 
+    bool shouldDumpConstantNodes = true;
+
+#ifdef CPU_DEBUG_CAPS
+    shouldDumpConstantNodes = !(graph.getProperty().debugCaps.shouldDumpConstNodes == "NO");
+#endif
+
     auto get_inputs = [&] (const MKLDNNNodePtr & node) {
         auto pr_edges = node->getParentEdges();
```
Comment (EgorDuplensky) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp:129]: could you please replace "auto" with "const auto&" for this lambda where it is applicable.

File: inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/dump_constant_layers.hpp [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <pugixml.hpp>
+#include "gtest/gtest.h"
+
+namespace ExecutionGraphTests {
+
+class ExecGraphDumpConstantLayers : public testing::TestWithParam<std::string> {
```
Comment (EgorDuplensky) [inline inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/dump_constant_layers.hpp:10]: I think you have already discussed it but just wondering if we really need tests for this debug feature?

File: inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp [hunk 3]
```diff
@@ -119,22 +119,34 @@ InferenceEngine::CNNNetwork dump_graph_as_ie_ngraph_net(const MKLDNNGraph &graph
...
+            }
+
             IE_ASSERT(node2layer.count(pr_node) == 1);
             auto pr = node2layer[pr_node];
 
-            inputs[ch_port] = pr->output(pr_port);
+            inputs[ch_port++] = pr->output(pr_port);
         }
-
+        inputs.resize(ch_port);
```
[the remark this answers is not in the document]
Comment (EgorDuplensky) [reply inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp:149]: Better to be discussed anyway

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp [hunk 4]
```diff
@@ -920,6 +920,12 @@ void MKLDNNGraph::SortTopologically() {
 }
 
 void MKLDNNGraph::GetPerfData(std::map<std::string, InferenceEngine::InferenceEngineProfileInfo> &perfMap) const {
+    bool shouldPrintConstantNodes = true;
+
+#ifdef CPU_DEBUG_CAPS
+    shouldPrintConstantNodes = !(getProperty().debugCaps.shouldDumpConstNodes == "NO");
```
Comment (EgorDuplensky) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp:926]: From my opinion getProperty() is a misname (especially considering what setProperty does)
I know that this is not directly related to your changes, but I propose to replace getProperty() with getConfig() everywhere (getConfig also returns const Config& which seems more properly)

**Comment by egorshul** [comment]
Fixed in 8dedf008c9300c63f5d5e9e41497dfd4dc919fb9. @EgorDuplensky thank you!

File: inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/dump_constant_layers.hpp (hunk 2 above)
Comment (egorshul) [reply inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/dump_constant_layers.hpp:10]: I think we need tests for this debugging feature.
@dmitry-gorokhov what do you think?

Comment (dmitry-gorokhov) [reply inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/dump_constant_layers.hpp:10]: This test is intended to check that exec_graph doesn't contain const nodes by default. Executable graph - public funtionality which we need to validate.
Tests for debug capabilities - separate topic. Lets discuss not in bounds of this PR.

Comment (EgorDuplensky) [reply inference-engine/tests/functional/plugin/shared/include/execution_graph_tests/dump_constant_layers.hpp:10]: Basically the question about the test for the PR is not applicable anymore, since now it is default behavior (to not include constant nodes) and I agree that this better to be tested.

File: inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp [hunk 5]
```diff
@@ -120,21 +120,35 @@ InferenceEngine::CNNNetwork dump_graph_as_ie_ngraph_net(const MKLDNNGraph &graph
...
+            const auto& pr_node = edge->getParent();
 
             IE_ASSERT(node2layer.count(pr_node) == 1);
-            auto pr = node2layer[pr_node];
+            const auto& pr = node2layer[pr_node];
 
             inputs[ch_port] = pr->output(pr_port);
         }
 
+#ifdef CPU_DEBUG_CAPS
```
Comment (EgorDuplensky) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp:138]: After changing the contexts (now we want to not include constant nodes by default) I like more your first implementation then this :), since we cannot isolate DEBUG_CAPS piece of code anymore.
In you previous implementation you moved ch_port outside the loop, maybe we can just intputs.insert() instead of index access and do not preallocate memory for inputs?
With those comments I am just trying to make code less ugly, which is always the case when we use #ifdefs.

Comment (egorshul) [reply inference-engine/src/mkldnn_plugin/mkldnn_graph_dumper.cpp]: Done

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky  any comments from your side?

**Review by EgorDuplensky — APPROVED** [review]

**Comment by EgorDuplensky** [comment]
@dmitry-gorokhov  Nope, Approved.

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/activation.cpp [hunk 6]
```diff
@@ -58,6 +58,7 @@ class ActivationLayerCPUTest : public testing::WithParamInterface<ActivationLaye
         std::tie(basicParamsSet, cpuParams) = this->GetParam();
 
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+        configuration.insert({"DUMP_CONSTANT_NODES", CONFIG_VALUE(YES)});
```
Comment (EgorDuplensky) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/activation.cpp:61]: Just curious, why do we want it to be enabled by default in tests?
If there is the reason to do so, instead of updating every test, can we just add:
```c++
if (targetDevice = "CPU")
    configuration.insert({"DUMP_CONSTANT_NODES", CONFIG_VALUE(YES)});
```
To LoadNetwork from layer_test_utils.cpp

Comment (egorshul) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/activation.cpp:61]: We need all the nodes, otherwise we will get an error:
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/tests/functional/plugin/cpu/test_utils/cpu_test_utils.cpp#L138

Comment (EgorDuplensky) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/activation.cpp:61]: is it possible to make that check slightly more complex so we could avoid other changes?

File: inference-engine/src/plugin_api/cpp_interfaces/interface/ie_internal_plugin_config.hpp [hunk 7]
```diff
@@ -45,6 +45,12 @@ DECLARE_CONFIG_KEY(CPU_THREADS_PER_STREAM);
  */
 DECLARE_CONFIG_KEY(FORCE_DISABLE_CACHE);
 
+/**
+ * @brief This key determines if constant nodes should be dumped
+ * @ingroup ie_dev_api_plugin_api
+ */
+DECLARE_CONFIG_KEY(DUMP_CONSTANT_NODES);
```
Comment (ilya-lavrenov) [inline inference-engine/src/plugin_api/cpp_interfaces/interface/ie_internal_plugin_config.hpp:52]: Why this cannot be enabled by default without an explicit option?

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/activation.cpp (hunk 6 above)
Comment (dmitry-gorokhov) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/activation.cpp:61]: @egorshul we shouldn't update all tests with logic like this: `configuration.insert({"DUMP_CONSTANT_NODES", CONFIG_VALUE(YES)});`. Lets do this in base class.

File: inference-engine/src/plugin_api/cpp_interfaces/interface/ie_internal_plugin_config.hpp (hunk 7 above)
Comment (dmitry-gorokhov) [reply inference-engine/src/plugin_api/cpp_interfaces/interface/ie_internal_plugin_config.hpp:52]: Actually it is enabled by default in master right mow. The intention is to avoid dumping constant nodes both in performance counters and execution graph by default to make them more readable. However there still might be cases there this information might be useful, so the option is added.
